### PR TITLE
Instrument pre-redirect flow

### PR DESCRIPTION
**Pull request by konstantin-msft**
- Instrument pre-redirect flow.

### CODE REVIEW

**Review by peterzenz — APPROVED** [review]
I think something to tie this back to the redirect when/if it lands on us so we can attempt to measure the redirect time might be useful.  Is there anything we can add to the reply URL, like a client timestamp, that would be preserved across the redirects?

**Comment by konstantin-msft** [comment]
> I think something to tie this back to the redirect when/if it lands on us so we can attempt to measure the redirect time might be useful. Is there anything we can add to the reply URL, like a client timestamp, that would be preserved across the redirects?

We can measure the redirect time by joining pre- and post- redirect events by correlation id and subtracting former end time from latter start time.

File: lib/msal-browser/src/controllers/StandardController.ts [hunk 1]
```diff
@@ -567,88 +567,117 @@ export class StandardController implements IController {
...
-                    }
-                    this.browserStorage.setInteractionInProgress(false);
-                    throw e;
-                });
-        } else {
-            const redirectClient = this.createRedirectClient(correlationId);
-            result = redirectClient.acquireToken(request);
-        }
+        BrowserUtils.redirectPreflightCheck(this.initialized, this.config);
+        this.browserStorage.setInteractionInProgress(true);
```
Comment (tnorling) [inline lib/msal-browser/src/controllers/StandardController.ts]: Can we move this back up to the top? This should ideally be the very first thing we do.

File: lib/msal-browser/src/controllers/StandardController.ts [hunk 2]
```diff
@@ -567,88 +567,117 @@ export class StandardController implements IController {
...
+                        EventType.LOGIN_FAILURE,
+                        InteractionType.Redirect,
+                        null,
+                        e
+                    );
+                }
+                throw e;
+            });
+        } catch (e) {
+            atrMeasurement.end({ success: false }, e);
```
Comment (tnorling) [inline lib/msal-browser/src/controllers/StandardController.ts:660]: Can we move the failure events down into this block?

**Review by tnorling — APPROVED** [review]

File: lib/msal-browser/src/controllers/StandardController.ts (hunk 1 above)
Comment (konstantin-msft) [reply lib/msal-browser/src/controllers/StandardController.ts]: I put it after the perf event initialization on purpose to instrument pre-flight errors similar to other APIs.

File: lib/msal-browser/src/controllers/StandardController.ts (hunk 2 above)
Comment (konstantin-msft) [reply lib/msal-browser/src/controllers/StandardController.ts:660]: Done

**Pull request merged**